citrination_client/client.py

```diff
@@ -271,19 +271,78 @@ def upload_file(self, file_path, data_set_id, root_path=None):
+        else:
+            return json.dumps({
+                    "message": "No file at specified path {0}".format(source_path)
+                })
+
+    def upload_file(self, file_path, dataset_id, root_path=None):
+        """
+        Upload file to Citrination.
+        :param file_path: File path to upload.
+        :param data_set_id: The dataset id to upload the file to.
+        :return: Response object or return code if the file was not uploaded.
+        """
+        return self.upload(dataset_id, file_path)
+
+    def list(self, dataset_id, glob=".", is_dir=False):
+        """
+        List matched filenames in a dataset on Citrination.
+        :param dataset_id: The ID of the dataset to search for files.
+        :param glob: A pattern which will be matched against files in the dataset.
+        :param is_dir: A boolean indicating whether or not the pattern should match against the beginning of paths in the dataset.
+        :return: Response object or return code if the file was not uploaded.
+        """
+        url = self.api_url + '/datasets/' + str(dataset_id) + '/list_filepaths'
+        data = {
+            "list": {
+                "glob": glob,
+                "is_dir": is_dir
+            }
+        }
+        r = requests.post(url, data=json.dumps(data), headers=self.headers)
+        return json.loads(r.content.decode('utf-8'))
+
+    def matched_file_count(self, dataset_id, glob=".", is_dir=False):
+        list_result = self.list(dataset_id, glob, is_dir)
+        if isinstance(list_result["files"], list):
+            return len(list_result["files"])
+        else:
+            return list_result
+
+    def get_matched_dataset_files(self, dataset_id, glob, is_dir=False, latest_version_only=True):
+        """
+        Retrieves URLs for the files matched by a glob or a path to a directory
+        in a given dataset.
+
+        :param dataset_id: The id of the dataset to retrieve files from
+        :param glob: A regex used to select one or more files in the dataset
+        :param is_dir: A flag used to indicate that the glob should be matched against the start of the paths in the dataset (simulates directory matching)
+        :param latest: A boolean flag indicating that results should be limited to reporting files from the latest dataset version
+        :return: The response object, or an error message object if the request failed
+        """
+        data = {
+            "download_request": {
+                "glob": glob,
+                "is_dir": is_dir,
+                "latest": latest_version_only
+            }
+        }
+        url = self.api_url + '/datasets/' + str(dataset_id) + '/download_files'
+        r = requests.post(url,data=json.dumps(data), headers=self.headers)
+        return json.loads(r.content.decode('utf-8'))
+
 
-    def get_dataset_files(self, dataset_id, latest = False):
+    def get_dataset_files(self, dataset_id, latest_version_only = False):
```

Comment:
I thinking more about scripts that would be calling this method _outside_ of this package. If those scripts are referring to `latest` then they would break, right?